test/inspector/inspector-helper-new.js

```diff
@@ -0,0 +1,427 @@
+    this.unprocessedStderrLines_ = [];
+
+    this.process_.stdout.on('data', makeBufferingDataCallback(
+      (line) => console.log('[out]', line)));
+
+    this.process_.stderr.on('data', makeBufferingDataCallback(
+      (message) => this.onStderrLine(message)));
+
+    this.shutdownPromise_ = new Promise((resolve) => {
+      this.process_.on('exit', (exitCode, signal) => {
+        resolve({ exitCode, signal });
+        this.running_ = false;
+      });
+    });
+  }
+
+  static createChildProcess(inspectorFlags, scriptContents, scriptFile) {
+    const args = [].concat(inspectorFlags);
+    if (scriptContents) {
+      args.push('-e', scriptContents);
+    } else {
+      args.push(scriptFile);
+    }
+
+    const child = spawn(process.execPath, args);
+
+    const handler = tearDown.bind(null, child);
+    process.on('exit', handler);
+    process.on('uncaughtException', handler);
+    process.on('unhandledRejection', handler);
+    process.on('SIGINT', handler);
+
+    return child;
+  }
+
+  onStderrLine(line) {
+    console.log('[err]', line);
+    if (this.portCallback_) {
+      const [ match, port ] =
+        line.match(/Debugger listening on ws:\/\/.+:(\d+)\/.+/);
+      if (match)
+        this.portCallback_(port);
+      this.portCallback_ = null;
+    }
+    if (this.stderrLineCallback_) {
+      this.stderrLineCallback_(line);
+      this.stderrLineCallback_ = null;
+    } else {
+      this.unprocessedStderrLines_.push(line);
+    }
+  }
+
+  assertDone() {
+    console.log(1);
+    assert.strictEqual(this.running_, false, 'Child instance still running');
+    console.log(2);
+  }
+
+  async testHttpResponse(host, path) {
+    console.log('[test]', `Testing ${path}`);
+    const response = await checkHttpResponse(host, await this.portPromise,
+                                             path);
+    try {
+      return JSON.parse(response);
+    } catch (e) {
+      e.response = response;
+      throw e;
+    }
+  }
+
+  wsHandshake(devtoolsUrl) {
+    return this.portPromise.then((port) => new Promise((resolve) => {
+      http.get({
+        port,
+        path: url.parse(devtoolsUrl).path,
+        headers: {
+          'Connection': 'Upgrade',
+          'Upgrade': 'websocket',
+          'Sec-WebSocket-Version': 13,
+          'Sec-WebSocket-Key': 'key=='
+        }
+      }).on('upgrade', (message, socket) => {
+        resolve(new InspectorSession(socket, this));
+      }).on('response', common.mustNotCall('Upgrade was not received'));
+    }));
+  }
+
+  async connectInspectorSession() {
+    console.log('[test]', 'Connecting to a child Node process');
+    const port = await this.portPromise;
+    const response = await checkHttpResponse(null, port, '/json/list');
+    const url = JSON.parse(response)[0]['webSocketDebuggerUrl'];
+    return await this.wsHandshake(url);
+  }
+
+  expectShutdown() {
+    return this.shutdownPromise_;
+  }
+
+  kill() {
+    return this.enqueue_((callback) => {
```

Comment:
I deleted "kill" (for now). It was copied from the old test and "enqueu_" is a primary reason the tests have to be rewritten :)